### DISCUSSION TITLE
Compilation problem

### DISCUSSION THREAD

**Discussion by thorfif** [EtoTen/stm32_can_sim#5] [General]
Hello,
thank you for your project.
However, I have a problem, when I try to compile I get the following errors. Could you please help

In file included from src\main.cpp:43:
.pio\libdeps\bluepill_f103c6\eXoCAN/eXoCAN.h:146:41: error: 'u_int32_t' has not been declared
  146 |   void filter32Init(int bank, int mode, u_int32_t a, u_int32_t b);                   //32b filters
      |                                         ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN/eXoCAN.h:146:54: error: 'u_int32_t' has not been declared
  146 |   void filter32Init(int bank, int mode, u_int32_t a, u_int32_t b);                   //32b filters
      |                                                      ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN/eXoCAN.h:158:35: error: 'u_int32_t' has not been declared
  158 |   void filterMask32Init(int bank, u_int32_t id = 0, u_int32_t mask = 0);
      |                                   ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN/eXoCAN.h:158:53: error: 'u_int32_t' has not been declared
  158 |   void filterMask32Init(int bank, u_int32_t id = 0, u_int32_t mask = 0);
      |                                                     ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN/eXoCAN.h:159:35: error: 'u_int32_t' has not been declared
  159 |   void filterList32Init(int bank, u_int32_t idA = 0, u_int32_t idB = 0); // 32b filters
      |                                   ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN/eXoCAN.h:159:54: error: 'u_int32_t' has not been declared
  159 |   void filterList32Init(int bank, u_int32_t idA = 0, u_int32_t idB = 0); // 32b filters
      |                                                      ^~~~~~~~~
In file included from .pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.cpp:1:
.pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.h:146:41: error: 'u_int32_t' has not been declared
  146 |   void filter32Init(int bank, int mode, u_int32_t a, u_int32_t b);                   //32b filters
      |                                         ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.h:146:54: error: 'u_int32_t' has not been declared
  146 |   void filter32Init(int bank, int mode, u_int32_t a, u_int32_t b);                   //32b filters
      |                                                      ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.h:158:35: error: 'u_int32_t' has not been declared
  158 |   void filterMask32Init(int bank, u_int32_t id = 0, u_int32_t mask = 0);
      |                                   ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.h:158:53: error: 'u_int32_t' has not been declared
  158 |   void filterMask32Init(int bank, u_int32_t id = 0, u_int32_t mask = 0);
      |                                                     ^~~~~~~~~
Compiling .pio\build\bluepill_f103c6\FrameworkArduino\VirtIOSerial.cpp.o
.pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.h:159:35: error: 'u_int32_t' has not been declared
  159 |   void filterList32Init(int bank, u_int32_t idA = 0, u_int32_t idB = 0); // 32b filters
      |                                   ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.h:159:54: error: 'u_int32_t' has not been declared
  159 |   void filterList32Init(int bank, u_int32_t idA = 0, u_int32_t idB = 0); // 32b filters
      |                                                      ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.cpp:129:41: error: 'u_int32_t' has not been declared
  129 | void eXoCAN::filterList32Init(int bank, u_int32_t idA, u_int32_t idB) //32b filters
      |                                         ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.cpp:129:56: error: 'u_int32_t' has not been declared
  129 | void eXoCAN::filterList32Init(int bank, u_int32_t idA, u_int32_t idB) //32b filters
      |                                                        ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.cpp:135:41: error: 'u_int32_t' has not been declared
  135 | void eXoCAN::filterMask32Init(int bank, u_int32_t id, u_int32_t mask) //32b filters
      |                                         ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.cpp:135:55: error: 'u_int32_t' has not been declared
  135 | void eXoCAN::filterMask32Init(int bank, u_int32_t id, u_int32_t mask) //32b filters
      |                                                       ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.cpp:140:47: error: 'u_int32_t' has not been declared
  140 | void eXoCAN::filter32Init(int bank, int mode, u_int32_t a, u_int32_t b) //32b filters
      |                                               ^~~~~~~~~
.pio\libdeps\bluepill_f103c6\eXoCAN\eXoCAN.cpp:140:60: error: 'u_int32_t' has not been declared
  140 | void eXoCAN::filter32Init(int bank, int mode, u_int32_t a, u_int32_t b) //32b filters
      |                                                            ^~~~~~~~~
*** [.pio\build\bluepill_f103c6\src\main.cpp.o] Error 1
*** [.pio\build\bluepill_f103c6\lib89b\eXoCAN\eXoCAN.cpp.o] Error 1

**Comment by EtoTen** [comment]
Weird, some dependancy must have changed and broken it.  I just tried to compile and got the same error. Let me dig into it. 

You should file an issue with https://github.com/exothink/eXoCAN too.

**Reply by EtoTen** [reply]
Fix:
open the #include <eXoCAN.h>
add this to the top of it, after "#include <arduino.h>", save, recompile

#include <cstdint>
#define u_int8_t uint8_t
#define u_int32_t uint32_t

**Reply by EtoTen** [reply]
update: I created issue with eXoCAN and a pull request
https://github.com/exothink/eXoCAN/pull/19
https://github.com/exothink/eXoCAN/issues/20

**Comment by thorfif** [comment]
Perfect. Thank You. Now compilation OK.
I'll have the SYNC 3 unit next week and test it out :)

Thank you very much for your help
